File: leetcode2138.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        vector<string>ans;
        for(int i = 0;i<s.length();i += k){
          int j = 0;
          string p;
          p = "";
          while(j < k){
            if(i + j < s.length()){
              p += s[i + j];
            }else{
              p += fill;
            }
            j ++;
          }
          ans.push_back(p);
        }
      return ans;
    }
};

File: leetcode2138.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
      ans: List[str] = []
      size: int = len(s)
      for i in range(0,size,k):
        if i + k <= size:
          ans.append(s[i:i + k])
        else:
          ans.append(s[i:] + (k - size%k) * fill)
      return ans
        
